galaxy/api/internal/views/imports.py

```diff
@@ -0,0 +1,97 @@
+# (c) 2012-2019, Ansible by Red Hat
+#
+# This file is part of Ansible Galaxy
+#
+# Ansible Galaxy is free software: you can redistribute it and/or modify
+# it under the terms of the Apache License as published by
+# the Apache Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# Ansible Galaxy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# Apache License for more details.
+#
+# You should have received a copy of the Apache License
+
+import collections
+
+from django.db.models import fields
+from django.db.models import F, Value
+
+from rest_framework import generics
+
+
+from galaxy.main import models
+from galaxy.api.internal.serializers.imports import (
+    TYPE_REPOSITORY, TYPE_COLLECTION, SERIALIZER_BY_TYPE
+)
+
+__all__ = (
+    'NamespaceImportsList',
+)
+
+
+class NamespaceImportsList(generics.ListAPIView):
+
+    QS_BY_TYPE = {
+        TYPE_COLLECTION: (
+            models.CollectionImport.objects
+            .select_related('pulp_task', 'namespace')
+            .all()
+        ),
+        TYPE_REPOSITORY: (
+            models.ImportTask.objects
+            .select_related('repository__provider_namespace__namespace')
+            .all()
+        ),
+    }
+
+    def list(self, request, *args, **kwargs):
+        items = self.paginate_queryset(self.get_queryset())
+        self.load_objects(items)
+        result = self.serialize_objects(items)
+        return self.get_paginated_response(result)
+
+    def get_queryset(self):
```

Comment:
Can you add parameters to filter by type (collection | repository) and name?

Reply:
Sure